### ISSUE TITLE
[Question] Stochastic Weight Averaging (SWA) in stable-baseline3 not RL Baselines3 Zoo

### ISSUE THREAD

**Issue by hontimzam** [DLR-RM/stable-baselines3#2149]
### ❓ Question

Hello, I am **Tim**. I am using the **A2C** model in **RL**. From my experience, I feel that the **SB3** is extremely user-friendly, we can implement a customise network, optimizer, etc in the **policy_kwargs**. However, 
I want to ask if we can easily add the **Stochastic Weight Averaging** (**SWA**) for the optimizer in the code? Like, only add a
policy_kwargs = {
'optimizer_class': th.optim.SWA,
'optimizer_kwargs': xxxxx}
From the **PyTorch**: https://docs.pytorch.org/docs/stable/optim.html#stochastic-weight-averaging
As I am not sure if it works because when I tried to study the train() function in **A2C**, I don't see a required function that is needed for **SWA**, as I see in the **PyTorch** document, it requires us to implement as below:

>>> loader, optimizer, model, loss_fn = ...
>>> swa_model = torch.optim.swa_utils.AveragedModel(model)
>>> scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=300)
>>> swa_start = 160
>>> swa_scheduler = SWALR(optimizer, swa_lr=0.05)
>>> for epoch in range(300):
>>>       for input, target in loader:
>>>           optimizer.zero_grad()
>>>           loss_fn(model(input), target).backward()
>>>           optimizer.step()
>>>       if epoch > swa_start:
>>>           swa_model.update_parameters(model)
>>>           swa_scheduler.step()
>>>       else:
>>>           scheduler.step()
>>> # Update bn statistics for the swa_model at the end
>>> torch.optim.swa_utils.update_bn(loader, swa_model)
>>> # Use swa_model to make predictions on test data
>>> preds = swa_model(test_input)

I do not see similar things on **A2C** - train() function. I also do not see any discussion about this in the stable-baseline3 **GitHub** issue forum. 

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [x] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [x] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.
